commit 55370810999a952ed2575cf959c6bf42573ac83b (refs/changes/26/3526/5)
Author: Angela Fong <angela.fong@joyent.com>
Date:   2018-04-24T22:09:14+00:00 (1 year, 5 months ago)
    
    TRITON-201 Docker doc: Divergence sections need update
    Reviewed by: Josh Wilsdon <jwilsdon@joyent.com>
    Approved by: Josh Wilsdon <jwilsdon@joyent.com>

diff --git a/docs/api/README.md b/docs/api/README.md
index f517df7..564506b 100644
--- a/docs/api/README.md
+++ b/docs/api/README.md
@@ -150,11 +150,13 @@ SDC Docker offers a number of features unique to Triton's container-native infra
 
 ## Docker CLI commands and Docker Remote API methods
 
-`docker attach`, `docker build`, `docker commit`, `docker cp`, `docker create`, `docker exec`,
-`docker history`, `docker images`, `docker info`, `docker inspect`, `docker kill`, `docker login`,
-`docker logout`, `docker logs`, `docker port`, `docker ps`, `docker pull`, `docker rename`,
-`docker restart`, `docker rm`, `docker rmi`, `docker run`, `docker search`, `docker start`,
-`docker stop`, `docker tag`, `docker top`, `docker version`, `docker wait`
+`docker attach`, `docker build`, `docker commit`, `docker cp`, `docker create`,
+`docker exec`, `docker history`, `docker images`, `docker info`, `docker inspect`,
+`docker kill`, `docker login`, `docker logout`, `docker logs`, `docker port`,
+`docker ps`, `docker pull`, `docker push`, `docker rename`, `docker restart`,
+`docker rm`, `docker rmi`, `docker run`, `docker search`, `docker start`,
+`docker stop`, `docker tag`, `docker top`, `docker version`, `docker volume`,
+`docker wait`
 
 ## Divergence
 
diff --git a/docs/api/commands/attach.md b/docs/api/commands/attach.md
index 58093c5..695a15b 100644
--- a/docs/api/commands/attach.md
+++ b/docs/api/commands/attach.md
@@ -4,6 +4,7 @@
 
     Attach to a running container
 
+      --detach-keys       Override the key sequence for detaching a container
       --no-stdin=false    Do not attach STDIN
       --sig-proxy=true    Proxy all received signals to the process
 
@@ -79,7 +80,7 @@ is returned by the `docker attach` command to its caller too:
 
 ## Divergence
 
-There is no known divergence between the Triton SDC Docker and Docker Inc. implementations of this method. Please contact Joyent support or file a ticket if you discover any.
+- `--detach-keys` option is unsupported.
 
 ## Related
 
diff --git a/docs/api/commands/build.md b/docs/api/commands/build.md
index b774f4a..5d2dd7a 100644
--- a/docs/api/commands/build.md
+++ b/docs/api/commands/build.md
@@ -15,6 +15,7 @@
       -f, --file=""                   Name of the Dockerfile (Default is 'PATH/Dockerfile')
       --force-rm                      Always remove intermediate containers
       --help                          Print usage
+      --label                         Set metadata for an image (default [])
       --isolation=""                  Container isolation technology
       -m, --memory=""                 Memory limit for all build containers
       --memory-swap=""                A positive integer equal to memory plus swap. Specify -1 to enable unlimited swap.
@@ -288,6 +289,7 @@ unique to Triton.
 * `--cpuset-cpus` and `--cpuset-mems` (controls which CPUs and memory nodes to run on) are ignored. See [resource allocation](../features/resources.md).
 * `--cpu-period` and `--cpu-quota` (limit the CPU CFS settings) are ignored. See [resource allocation](../features/resources.md).
 * `--isolation` is unsupported.
+* `--label` is unimplemented.
 * `--quiet` is unimplemented.
 * `--shm-size` is ignored.
 * `--ulimit` (ulimit options) is unsupported.
@@ -302,3 +304,4 @@ unique to Triton.
 
 - [`docker commit`](../commands/commit.md)
 - [`docker images`](../commands/images.md)
+- [`docker push`](../commands/push.md)
diff --git a/docs/api/commands/commit.md b/docs/api/commands/commit.md
index 2230365..fb4fbe4 100644
--- a/docs/api/commands/commit.md
+++ b/docs/api/commands/commit.md
@@ -81,3 +81,4 @@ if you discover any.
 
 - [`docker build`](../commands/build.md)
 - [`docker images`](../commands/images.md)
+- [`docker push`](../commands/push.md)
diff --git a/docs/api/commands/cp.md b/docs/api/commands/cp.md
index 2f06118..98e98ed 100644
--- a/docs/api/commands/cp.md
+++ b/docs/api/commands/cp.md
@@ -6,6 +6,7 @@ Copy files/folders between a container and the local filesystem.
             docker cp [options] LOCALPATH|- CONTAINER:PATH
 
     --help  Print usage statement
+    -L, --follow-link   Always follow symbol link in SRC_PATH
 
 In the first synopsis form, the `docker cp` utility copies the contents of `PATH`
 from the filesystem of `CONTAINER` to the `LOCALPATH` (or stream as a tar archive
@@ -76,9 +77,7 @@ of the resource from the source container as a tar archive to `STDOUT`.
 
 ## Divergence
 
-With sdc-docker, `docker cp` does not work against stopped containers. ([DOCKER-374](http://smartos.org/bugview/DOCKER-374))
-
-Copying of files from local path to container path is not supported. Follow [DOCKER-527](http://smartos.org/bugview/DOCKER-527) for updates.
+- `-L` and --follow-link` options are unsupported.
 
 ## Related
 
diff --git a/docs/api/commands/create.md b/docs/api/commands/create.md
index 60257dc..bef14ec 100644
--- a/docs/api/commands/create.md
+++ b/docs/api/commands/create.md
@@ -9,6 +9,7 @@ Creates a new container.
       -a, --attach=[]                  Attach to STDIN, STDOUT or STDERR
       --add-host=[]                    Add a custom host-to-IP mapping (host:ip)
       --blkio-weight=0                 Block IO weight (relative weight)
+      --blkio-weight-device            Block IO weight (relative device weight) (default [])
       -c, --cpu-shares=0               CPU shares (relative weight)
       --cap-add=[]                     Add Linux capabilities
       --cap-drop=[]                    Drop Linux capabilities
@@ -16,10 +17,13 @@ Creates a new container.
       --cidfile=""                     Write the container ID to the file
       --cpuset-cpus=""                 CPUs in which to allow execution (0-3, 0,1)
       --cpuset-mems=""                 Memory nodes (MEMs) in which to allow execution (0-3, 0,1)
+      --cpu-count                      CPU count (Windows only)
+      --cpu-percent                    CPU percent (Windows only)
       --cpu-period=0                   Limit the CPU CFS (Completely Fair Scheduler) period
       --cpu-quota=0                    Limit the CPU CFS (Completely Fair Scheduler) quota
-      --detach-keys                    Override the key sequence for detaching a container
+      --credentialspec                 Credential spec for managed service account (Windows only)
       --device=[]                      Add a host device to the container
+      --device-cgroup-rule=[]          Add a rule to the cgroup allowed devices list (default [])
       --device-read-bps=[]             Limit read rate (bytes per second) from a device
       --device-read-iops=[]            Limit read rate (IO per second) from a device
       --device-write-bps=[]            Limit write rate (bytes per second) to a device
@@ -32,15 +36,23 @@ Creates a new container.
       --entrypoint=""                  Overwrite the default ENTRYPOINT of the image
       --env-file=[]                    Read in a file of environment variables
       --expose=[]                      Expose a port or a range of ports
+      --group-add                      Add additional groups to join (default [])
+      --health-cmd                     Command to run to check health
+      --health-interval                Time between running the check
+      --health-retries                 Consecutive failures needed to report unhealthy
+      --health-timeout                 Maximum time to allow one check to run
       -h, --hostname=""                Container host name
       --help=false                     Print usage
       -i, --interactive=false          Keep STDIN open even if not attached
+      --io-maxbandwidth                Maximum IO bandwidth limit for the system drive (Windows only)
+      --io-maxiops                     Maximum IOps limit for the system drive (Windows only)
       --ip                             Container IPv4 address (e.g. 172.30.100.104)
       --ip6                            Container IPv6 address (e.g. 2001:db8::33)
       --ipc=""                         IPC namespace to use
       --isolation                      Container isolation level
       --kernel-memory                  Kernel memory limit
       --link=[]                        Add link to another container
+      --link-local-ip                  Container IPv4/IPv6 link-local addresses (default [])
       --log-driver=""                  Logging driver for container
       --log-opt=[]                     Log driver specific options
       -l, --label=[]                   Set metadata on the container (e.g., --label=com.example.key=value)
@@ -49,30 +61,38 @@ Creates a new container.
       --mac-address=""                 Container MAC address (e.g. 92:d0:c6:0a:29:33)
       --memory-reservation             Memory soft limit
       --memory-swap=""                 Total memory (memory + swap), '-1' to disable swap
-      --memory-swappiness=""           Tune a container's memory swappiness behavior. Accepts an integer between 0 and 100
+      --memory-swappiness=""           Tune container memory swappiness (0 to 100)
       --name=""                        Assign a name to the container
       --network=default                Connect a container to a network
       --network-alias=[]               Add network-scoped alias for the container
+      --no-healthcheck                 Disable any container-specified HEALTHCHECK
       --oom-kill-disable=false         Whether to disable OOM Killer for the container or not
       --oom-score-adj                  Tune host's OOM preferences (-1000 to 1000)
       -P, --publish-all=false          Publish all exposed ports to random ports
       -p, --publish=[]                 Publish a container's port(s) to the host
       --pid=""                         PID namespace to use
+      --pids-limit                     Tune container pids limit (set -1 for unlimited)
       --privileged=false               Give extended privileges to this container
       --read-only=false                Mount the container's root filesystem as read only
       --restart="no"                   Restart policy (no, on-failure[:max-retry], always)
+      --runtime                        Runtime to use for this container
       --security-opt=[]                Security Options
       --shm-size                       Size of /dev/shm, default value is 64MB
+      --stop-signal=SIGTERM            Signal to stop a container, SIGTERM by default
+      --storage-opt                    Storage driver options for the container (default [])
+      --sysctl                         Sysctl options (default map[])
       -t, --tty=false                  Allocate a pseudo-TTY
       --tmpfs=[]                       Mount a tmpfs directory
       -u, --user=""                    Username or UID (format: <name|uid>[:<group|gid>])
       --ulimit=[]                      Ulimit options
+      --userns                         User namespace to use
       --uts=""                         UTS namespace to use
       -v, --volume=[]                  Bind mount a volume
       --volume-driver                  Optional volume driver for the container
       --volumes-from=[]                Mount volumes from the specified container(s)
       -w, --workdir=""                 Working directory inside the container
 
+
 The `docker create` command creates a writeable container layer over
 the specified image and prepares it for running the specified command.
 The container ID is then printed to `STDOUT`.
@@ -131,42 +151,50 @@ features unique to Triton. See the [Resources](../features/resources.md) and
 [Networks](../features/networks.md) documentation for more information on how
 to size containers and assign public or private IP addresses to them.
 
-* `--add-host` (host-to-IP mapping) has been implemented in
-  [DOCKER-694](https://smartos.org/bugview/DOCKER-694). However, until
-  [DOCKER-775](https://smartos.org/bugview/DOCKER-775) is implemented and
-  deployed, this option will not work reliably until all compute nodes used for
-  Docker containers have been updated to a platform greater than 20160331.
-* `--blkio-weight` (block IO weight) is unsupported.
+Windows-based images and the associated parameters are not supported on Triton.
+
+* `--add-host` (host-to-IP mapping) option is supported on platform image
+  version on or after 20160331.
+* `--blkio-weight`, `blkio-weight-device` options are unsupported.
 * `--cgroup-parent` is ignored. See [security](../features/security.md).
-* `--cpu-shares` and `-c` are ignored, though CPU resources can be specified in conjunction with RAM. See [resource allocation](../features/resources.md). Joyent is working with the Docker community to improve how CPU resources are specified in the API.
+* `--cpu-percent`, `--cpu-shares` and `-c` are ignored, though CPU resources can
+  be specified in conjunction with RAM. See [resource allocation](../features/resources.md).
 * `--cap-add` and `--cap-drop` (Linux capabilities) are ignored. See [security](../features/security.md).
 * `--cpuset-cpus` and `--cpuset-mems` (controls which CPUs and memory nodes to run on) are ignored. See [resource allocation](../features/resources.md).
 * `--cpu-period` and `--cpu-quota` (limit the CPU CFS settings) are ignored. See [resource allocation](../features/resources.md).
-* `--detach-keys` is unsupported.
-* `--device` (mounts host device into container) is ignored.
+* `--device` and `--device-cgroup-rule` (mounts host device into container) is ignored.
 * `--device-read`, `--device-write` (device read/write rate limits) are unsupported.
 * `--disable-content-trust` (skip image verification) is ignored, follow [DOCKER-531](http://smartos.org/bugview/DOCKER-531).
 * `--dns-opt` (DNS options) are unimplemented at this time.
 * `--group-add` is unsupported.
+* `--health-cmd`, `--no-healthcheck` and other healthcheck options are unsupported.
+* `--io-maxbandwidth` and `--io-maxiops` options are unsupported.
 * `--ipc` is ignored.
+* `--ip`, `--ip6`, `--link-local-ip` and `--network-alias` (controls for network
+  config and ip address assignment) are currently unsupported.
 * `--log-driver` and `--log-opt` work somewhat differently on sdc-docker. See [log drivers](../features/logdrivers.md).
 * `--mac-address` which is unsupported. See [networking](../features/networks.md).
 * `--kernel-memory` and `--memory-reservation` (memory limits) are unsupported. See [resource allocation](../features/resources.md) for more about memory allocation in Triton.
 * `--memory-swap` and `--memory-swappiness` (disabling swap and tuning memory swappiness) are unsupported.
-* `--network-alias`, `--ip`, `--ip6` (controls network config and ip address
   assignment) are currently unsupported.
 * `--oom-kill-disable` and `--oom-score-adj` (tunables for OOM behavior) are unsupported.
-* `--pid=host` is unsupported.
+* `--pid` and `--pids-limit` are unsupported.
 * `-P`, `--publish-all`, `-p`, and `--publish` behave slightly differently thanks to each container having a complete IP stack with one or more virtual NICs. See [networking](../features/networks.md).
 * `--privileged` (extended privileges for containers) is ignored. See [security](../features/security.md).
 * `--read-only` is currently unimplemented, follow [DOCKER-158](http://smartos.org/bugview/DOCKER-158) for updates.
+* `--runtime` is not supported.
 * `--security-opt` (security options) is unsupported.
 * `--shm-size` (size of /dev/shm) is unsupported.
+* `--storage-opt` (storage driver) is unsupported.
+* `--sysctl` options are unsupported.
 * `--tmpfs` (mounting of tmpfs directory) is currently unimplemented, follow [DOCKER-667](http://smartos.org/bugview/DOCKER-667) for updates.
 * `--ulimit` (ulimit options) is unsupported.
+* `--userns` (user namespace) is unsupported.
 * `--uts` (UTS namespace to use) is unsupported.
 * `-v`, `--volume` and `--volumes-from` behave slightly differently in a Triton's container-native environment. See [volumes](../features/volumes.md).
-* `--volume-driver` and other `docker volume` commands are unimplemented at this time.
+* `--volume-driver` currently supports `tritonnfs` volume driver only. See
+  [volumes](../features/volumes.md) for more details about the Triton's
+  NFS volume feature.
 
 Docker Swarm's affinity filters for controlling on which server a container
 is provisioned are supported. See the [placement feature
diff --git a/docs/api/commands/inspect.md b/docs/api/commands/inspect.md
index 528c3c6..07c9c89 100644
--- a/docs/api/commands/inspect.md
+++ b/docs/api/commands/inspect.md
@@ -5,6 +5,7 @@
     Return low-level information on a container or image
 
       -f, --format=""            Format the output using the given go template
+      -s, --size                 Display total file sizes if the type is container
       --type=container|image     Return JSON for specified type, permissible
                                  values are "image" or "container"
 
@@ -83,6 +84,8 @@ VM.
 
 Read more about container [security on Triton](features/security.md) and see [notes about exit status](../divergence.md).
 
+The `-s` and `--size` options are currently unsupported.
+
 Please contact Joyent support or file a ticket if you discover any additional divergence.
 
 ## Related
diff --git a/docs/api/commands/logs.md b/docs/api/commands/logs.md
index b689c6f..d9a0f7f 100644
--- a/docs/api/commands/logs.md
+++ b/docs/api/commands/logs.md
@@ -4,6 +4,7 @@
 
     Fetch the logs of a container
 
+      --details                 Show extra details provided to logs
       -f, --follow=false        Follow log output
       --since=""                Show logs since timestamp
       -t, --timestamps=false    Show timestamps
@@ -29,7 +30,7 @@ The --since option shows only the container logs generated after a given date. Y
 
 ## Divergence
 
-There is no known divergence between the Triton SDC Docker and Docker Inc. implementations of this method. Please contact Joyent support or file a ticket if you discover any.
+- `--details` option is unsupported.
 
 ## Related
 
diff --git a/docs/api/commands/push.md b/docs/api/commands/push.md
index bd1ace2..a4a4cc0 100644
--- a/docs/api/commands/push.md
+++ b/docs/api/commands/push.md
@@ -1,9 +1,17 @@
 # push
 
+    Usage: docker push [OPTIONS] NAME[:TAG]
+
+    Push an image or a repository to a registry
+
+      --disable-content-trust   Skip image signing (default true)
+      --help                    Print usage
+
 ## Divergence
 
-- This command is currently unimplemented. Follow [DOCKER-73](http://smartos.org/bugview/DOCKER-73) for progress updates.
+- This supports only docker v2 image format.
 
 ## Related
 
-- Insert a list of related Docker and CloudAPI methods here
+- [`docker build`](../commands/build.md)
+- [`docker commit`](../commands/commit.md)
diff --git a/docs/api/commands/run.md b/docs/api/commands/run.md
index 970be3e..0347b67 100644
--- a/docs/api/commands/run.md
+++ b/docs/api/commands/run.md
@@ -7,6 +7,7 @@
       -a, --attach=[]                  Attach to STDIN, STDOUT or STDERR
       --add-host=[]                    Add a custom host-to-IP mapping (host:ip)
       --blkio-weight=0                 Block IO weight (relative weight)
+      --blkio-weight-device            Block IO weight (relative device weight) (default [])
       -c, --cpu-shares=0               CPU shares (relative weight)
       --cap-add=[]                     Add Linux capabilities
       --cap-drop=[]                    Drop Linux capabilities
@@ -14,11 +15,15 @@
       --cidfile=""                     Write the container ID to the file
       --cpuset-cpus=""                 CPUs in which to allow execution (0-3, 0,1)
       --cpuset-mems=""                 Memory nodes (MEMs) in which to allow execution (0-3, 0,1)
+      --cpu-count                      CPU count (Windows only)
+      --cpu-percent                    CPU percent (Windows only)
       --cpu-period=0                   Limit the CPU CFS (Completely Fair Scheduler) period
       --cpu-quota=0                    Limit the CPU CFS (Completely Fair Scheduler) quota
+      --credentialspec                 Credential spec for managed service account (Windows only)
       -d, --detach=false               Run container in background and print container ID
       --detach-keys                    Override the key sequence for detaching a container
       --device=[]                      Add a host device to the container
+      --device-cgroup-rule=[]          Add a rule to the cgroup allowed devices list (default [])
       --device-read-bps=[]             Limit read rate (bytes per second) from a device
       --device-read-iops=[]            Limit read rate (IO per second) from a device
       --device-write-bps=[]            Limit write rate (bytes per second) to a device
@@ -31,15 +36,23 @@
       --entrypoint=""                  Overwrite the default ENTRYPOINT of the image
       --env-file=[]                    Read in a file of environment variables
       --expose=[]                      Expose a port or a range of ports
+      --group-add                      Add additional groups to join (default [])
+      --health-cmd                     Command to run to check health
+      --health-interval                Time between running the check
+      --health-retries                 Consecutive failures needed to report unhealthy
+      --health-timeout                 Maximum time to allow one check to run
       -h, --hostname=""                Container host name
       --help=false                     Print usage
       -i, --interactive=false          Keep STDIN open even if not attached
+      --io-maxbandwidth                Maximum IO bandwidth limit for the system drive (Windows only)
+      --io-maxiops                     Maximum IOps limit for the system drive (Windows only)
       --ip                             Container IPv4 address (e.g. 172.30.100.104)
       --ip6                            Container IPv6 address (e.g. 2001:db8::33)
       --ipc=""                         IPC namespace to use
       --isolation                      Container isolation level
       --kernel-memory                  Kernel memory limit
       --link=[]                        Add link to another container
+      --link-local-ip                  Container IPv4/IPv6 link-local addresses (default [])
       --log-driver=""                  Logging driver for container
       --log-opt=[]                     Log driver specific options
       -l, --label=[]                   Set metadata on the container (e.g., --label=com.example.key=value)
@@ -52,23 +65,29 @@
       --name=""                        Assign a name to the container
       --network=default                Connect a container to a network
       --network-alias=[]               Add network-scoped alias for the container
+      --no-healthcheck                 Disable any container-specified HEALTHCHECK
       --oom-kill-disable=false         Whether to disable OOM Killer for the container or not
       --oom-score-adj                  Tune host's OOM preferences (-1000 to 1000)
       -P, --publish-all=false          Publish all exposed ports to random ports
       -p, --publish=[]                 Publish a container's port(s) to the host
       --pid=""                         PID namespace to use
+      --pids-limit                     Tune container pids limit (set -1 for unlimited)
       --privileged=false               Give extended privileges to this container
       --read-only=false                Mount the container's root filesystem as read only
       --restart="no"                   Restart policy (no, on-failure[:max-retry], always)
       --rm=false                       Automatically remove the container when it exits
+      --runtime                        Runtime to use for this container
       --security-opt=[]                Security Options
       --shm-size                       Size of /dev/shm, default value is 64MB
       --sig-proxy=true                 Proxy received signals to the process
       --stop-signal=SIGTERM            Signal to stop a container, SIGTERM by default
+      --storage-opt                    Storage driver options for the container (default [])
+      --sysctl                         Sysctl options (default map[])
       -t, --tty=false                  Allocate a pseudo-TTY
       --tmpfs=[]                       Mount a tmpfs directory
       -u, --user=""                    Username or UID (format: <name|uid>[:<group|gid>])
       --ulimit=[]                      Ulimit options
+      --userns                         User namespace to use
       --uts=""                         UTS namespace to use
       -v, --volume=[]                  Bind mount a volume
       --volume-driver                  Optional volume driver for the container
@@ -480,42 +499,50 @@ features unique to Triton. See the [Resources](../features/resources.md) and
 [Networks](../features/networks.md) documentation for more information on how
 to size containers and assign public or private IP addresses to them.
 
-* `--add-host` (host-to-IP mapping) has been implemented in
-  [DOCKER-694](https://smartos.org/bugview/DOCKER-694). However, until
-  [DOCKER-775](https://smartos.org/bugview/DOCKER-775) is implemented and
-  deployed, this option will not work reliably until all compute nodes used for
-  Docker containers have been updated to a platform greater than 20160331.
-* `--blkio-weight` (block IO weight) is unsupported.
+Windows-based images and the associated parameters are not supported on Triton.
+
+* `--add-host` (host-to-IP mapping) option is supported on platform image
+  version on or after 20160331.
+* `--blkio-weight`, `blkio-weight-device` options are unsupported.
 * `--cgroup-parent` is ignored. See [security](../features/security.md).
-* `--cpu-shares` and `-c` are ignored, though CPU resources can be specified in conjunction with RAM. See [resource allocation](../features/resources.md). Joyent is working with the Docker community to improve how CPU resources are specified in the API.
+* `--cpu-percent`, `--cpu-shares` and `-c` are ignored, though CPU resources can
+  be specified in conjunction with RAM. See [resource allocation](../features/resources.md).
 * `--cap-add` and `--cap-drop` (Linux capabilities) are ignored. See [security](../features/security.md).
 * `--cpuset-cpus` and `--cpuset-mems` (controls which CPUs and memory nodes to run on) are ignored. See [resource allocation](../features/resources.md).
 * `--cpu-period` and `--cpu-quota` (limit the CPU CFS settings) are ignored. See [resource allocation](../features/resources.md).
 * `--detach-keys` is unsupported.
-* `--device` (mounts host device into container) is ignored.
+* `--device` and `--device-cgroup-rule` (mounts host device into container) is ignored.
 * `--device-read`, `--device-write` (device read/write rate limits) are unsupported.
 * `--disable-content-trust` (image verification) is ignored at this time, follow [DOCKER-531](http://smartos.org/bugview/DOCKER-531) for updates.
 * `--dns-opt` (DNS options) are unimplemented at this time.
 * `--group-add` is unsupported.
+* `--health-cmd`, `--no-healthcheck` and other healthcheck options are unsupported.
+* `--io-maxbandwidth` and `--io-maxiops` options are unsupported.
+* `--ip`, `--ip6`, `--link-local-ip` and `--network-alias` (controls for network
+  config and ip address assignment) are currently unsupported.
 * `--ipc` is ignored.
 * `--log-driver` and `--log-opt` work somewhat differently on sdc-docker. See [log drivers](../features/logdrivers.md).
 * `--mac-address` which is unsupported. See [networking](../features/networks.md).
 * `--kernel-memory` and `--memory-reservation` (memory limits) are unsupported. See [resource allocation](../features/resources.md) for more about memory allocation in Triton.
 * `--memory-swap` and `--memory-swappiness` (disabling swap and tuning memory swappiness) are unsupported.
-* `--network-alias`, `--ip`, `--ip6` (controls network config and ip address
-  assignment) are currently unsupported.
 * `--oom-kill-disable` and `--oom-score-adj` (tunables for OOM behavior) are unsupported.
-* `--pid=host` is unsupported.
+* `--pid` and `--pids-limit` are unsupported.
 * `-P`, `--publish-all`, `-p`, and `--publish` behave slightly differently thanks to each container having a complete IP stack with one or more virtual NICs. See [networking](../features/networks.md).
 * `--privileged` (extended privileges for containers) is ignored. See [security](../features/security.md).
 * `--read-only` is currently unimplemented, follow [DOCKER-158](http://smartos.org/bugview/DOCKER-158) for updates.
+* `--runtime` is not supported.
 * `--security-opt` (security options) is unsupported.
 * `--shm-size` (size of /dev/shm) is unsupported.
+* `--storage-opt` (storage driver) is unsupported.
+* `--sysctl` options are unsupported.
 * `--tmpfs` (mounting of tmpfs directory) is currently unimplemented, follow [DOCKER-667](http://smartos.org/bugview/DOCKER-667) for updates.
 * `--ulimit` (ulimit options) is unsupported.
+* `--userns` (user namespace) is unsupported.
 * `--uts` (UTS namespace to use) is unsupported.
 * `-v`, `--volume` and `--volumes-from` behave slightly differently in a Triton's container-native environment. See [volumes](../features/volumes.md).
-* `--volume-driver` and other `docker volume` commands are unimplemented at this time.
+* `--volume-driver` currently supports `tritonnfs` volume driver only. See
+  [volumes](../features/volumes.md) for more details about the Triton's
+  NFS volume feature.
 
 Docker Swarm's affinity filters for controlling on which server a container
 is provisioned are supported. See the [placement feature
diff --git a/docs/api/commands/search.md b/docs/api/commands/search.md
index 779ac1b..e028ac9 100644
--- a/docs/api/commands/search.md
+++ b/docs/api/commands/search.md
@@ -6,7 +6,8 @@ Search [Docker Hub](https://hub.docker.com) for images
 
     Search the Docker Hub for images
 
-      --automated=false    Only show automated builds
+      -f, --filter         Filter output based on conditions provided (default [])
+      --limit              Max number of search results (default 25)
       --no-trunc=false     Don't truncate output
       -s, --stars=0        Only displays with at least x stars
 
@@ -19,9 +20,7 @@ more details on finding shared images from the command line.
 
 ## Divergence
 
-There is no known divergence between the Triton SDC Docker and Docker Inc.
-implementations of this method. Please contact Joyent support or file a ticket
-if you discover any.
+- `-f`, `--filter` and `--limit` options are unsupported.
 
 ## Related
 
diff --git a/docs/api/commands/stats.md b/docs/api/commands/stats.md
index a845d38..a65291d 100644
--- a/docs/api/commands/stats.md
+++ b/docs/api/commands/stats.md
@@ -4,8 +4,9 @@ Usage: docker stats CONTAINER [CONTAINER...]
 
 Display a live stream of one or more containers' resource usage statistics
 
-  --help=false       Print usage
-  --no-stream=false  Disable streaming stats and only pull the first result
+    -a, --all          Show all containers (default shows just running)
+    --help=false       Print usage
+    --no-stream=false  Disable streaming stats and only pull the first result
 
 Example:
 
@@ -22,6 +23,9 @@ underlying docker stats (cgroups) data points are available, see:
 
 - [docker-stats.js](https://github.com/joyent/sdc-cn-agent/blob/master/bin/docker-stats.js)
 
+Also the `-a` and `--all` options are not supported.
+
 ## Related
 
-- Insert a list of related Docker and CloudAPI methods here
+- [Container Monitor 'cmon'](https://github.com/joyent/triton-cmon/tree/master/docs)
+- [Container Monitor with Prometheus](https://docs.joyent.com/public-cloud/api-access/prometheus)
diff --git a/docs/api/divergence.md b/docs/api/divergence.md
index a0b8b35..d56e276 100644
--- a/docs/api/divergence.md
+++ b/docs/api/divergence.md
@@ -87,15 +87,11 @@ Here's the list of API methods currently unimplemented as of this writing, but
 expect it to get shorter by the day:
 
 `docker diff`, `docker events`, `docker export`, `docker import`, `docker load`,
-`docker network`, `docker node`, `docker pause`, `docker push`, `docker save`,
-`docker service`, `docker swarm`, `docker unpause`, `docker update`, `docker volume`
+`docker network`, `docker node`, `docker pause`, `docker save`,
+`docker service`, `docker swarm`, `docker unpause`, `docker update`
 
 ### Roadmap:
-- `docker push` is close to completion, it will be available in a later release
-  in 2016.
-- `docker volume` is in active development, more [details](https://github.com/joyent/rfd/blob/master/rfd/0026/README.md)
-  here.
-- `docker network` is also on our near-term roadmap, follow 
+- `docker network` is on our near-term roadmap, follow
 [DOCKER-722](http://smartos.org/bugview/DOCKER-722), [DOCKER-723](http://smartos.org/bugview/DOCKER-723),
 [DOCKER-724](http://smartos.org/bugview/DOCKER-724), [DOCKER-725](http://smartos.org/bugview/DOCKER-725)
   for updates.
@@ -173,3 +169,10 @@ these label names are currently defined:
   package for the container. The value can be a package name like
   `g4-standard-1G`, a UUID, or the first 8 characters of a UUID
   (short-UUID).
+
+
+## Other Differences
+
+SDC Docker supports certain versions of Docker CLI and Compose clients. Check
+the [supported versions](../#docker-version) and the individual command
+references to see which command options can be used with SDC Docker.
